In [23]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import pearsonr



In [24]:

# Import data
dfgini = pd.read_csv("https://raw.githubusercontent.com/edebrouwert/Income-Inequality-GDP/main/economic-inequality-gini-index.csv")
dfgdp = pd.read_csv("https://raw.githubusercontent.com/edebrouwert/Income-Inequality-GDP/main/gdp-per-capita-maddison-2020.csv")


In [22]:


# Inner join data frames on columns Entity, Code and Year to get a merged data frame with column for GDP and a column for Gini
mergeddf = pd.merge(dfgdp, dfgini, on=["Entity", "Code", "Year"], how="inner")

# Drop unnecessary column
mergeddf.drop(columns="417485-annotations", inplace=True)
mergeddf

# Check correlation with np array
nparrGDP = np.array(mergeddf["GDP per capita"])
nparrGini = np.array(mergeddf["Gini coefficient"])

np.corrcoef(nparrGDP, nparrGini)

# Check p value 
PearsonRvalues = pearsonr(mergeddf["GDP per capita"], mergeddf["Gini coefficient"])




n         r           CI95%         p-val       BF10  power
pearson  1810 -0.432258  [-0.47, -0.39]  2.705272e-83  3.836e+79    1.0

The coefficient is -0.432258. The	 p value is 2.705272e-83. Therefore we can conclude that there is a siginificant negative lineair relationship between a country's GDP and it's Gini coefficient